components/count-badge/count-badge-icon.js

```diff
@@ -79,15 +77,32 @@ class CountBadgeIcon extends CountBadgeMixin(LitElement) {
 	}
 
 	render() {
-		const numberPadding = this.size === 'small' ? '0.55rem' : '0.7rem';
-		const numberStyles = {
+		let numberStyles = {
 			border: '2px solid white',
-			position: 'relative',
-			left: this.dir === 'rtl' ? 0 : 'var(--d2l-count-badge-icon-padding)',
-			right: this.dir === 'rtl' ? 'var(--d2l-count-badge-icon-padding)' : 0,
-			top: numberPadding,
-			visibility: this.skeleton ? 'hidden' : undefined
+			visibility: this.skeleton ? 'hidden' : undefined,
+			display: 'inline-block',
+			position: 'absolute',
+			width: 'max-content'
```

Comment:
Width: max-content allows the badge to overflow slightly outside the padding if required.